### DISCUSSION TITLE
NextJS SSR Rendering Prefetching queries - Is it possible to redirect 404 if prefetching does not return anything ?

### DISCUSSION THREAD

**Discussion by Jonashr** [TanStack/query#4115] [Q&A]
I have a page that runs on SSR in NextJS. This page is just simply called /[category]. Is it possible to redirect to the 404 page on ServerSideProps when you use preFetchQuery if the category does not exist ? Here is an example that I have tried:

```
export async function getServerSideProps(context) {
  const categoryName = context.params.category;
  const queryClient = new QueryClient();

  try {
    await queryClient.prefetchQuery(['category', categoryName], () => fetchCategory(categoryName));
  } catch (error) {
    return {
      notFound: true,
    };
  }

  return {
    props: {
      dehydratedState: dehydrate(queryClient)
    },
  };
}
```

However, the prefetch query returns null so this does not work. Which means the redirect has to happen after getServerSideProps runs, which I don't think is ideal.

If I replace the try/catch with the following it works:

```
  try {
    const category = await  fetchCategory(categoryName);
  } catch (error) {
   // If category does not exist, notFound will be set to true.
    return {
      notFound: true,
    };
  }
```

So my question is really, is there a way to redirect to 404 page using getServerSideProps when prefetching a query if no data is returned, or does this only work when not using prefetching ?

**Comment by TkDodo** [comment]
replace prefetchQuery with fetchQuery.
